### PR TITLE
visibleInShow closure

### PR DESCRIPTION
**Pull request by izica**
https://github.com/Laravel-Backpack/ideas/issues/7

I actively use show operation, ability to show/hide a data row based on model data is often very useful.

```php
        $this->crud->addColumns([
            [
                'label'         => 'Name',
                'name'          => 'name',
                'visibleInShow' => function ($entry) {
                    return !!$entry->name;
                },
            ],
        ]);
```

### CODE REVIEW

File: src/app/Http/Controllers/Operations/ShowOperation.php [hunk 1]
```diff
@@ -102,8 +102,10 @@ public function show($id)
             }
 
             // remove columns that have visibleInShow set as false
-            if (isset($column['visibleInShow']) && $column['visibleInShow'] == false) {
-                $this->crud->removeColumn($column['key']);
+            if (isset($column['visibleInShow'])) {
+                if ((is_callable($column['visibleInShow']) && $column['visibleInShow']($this->data['entry']) === false) || $column['visibleInShow'] == false) {
```
Comment (pxpm) [inline src/app/Http/Controllers/Operations/ShowOperation.php:106]:
```suggestion
                if ((is_callable($column['visibleInShow']) && $column['visibleInShow']($this->data['entry']) === false) || $column['visibleInShow'] === false) {
```

**Comment by tabacitu** [comment]
Hmm... @izica sorry for leaving this hanging for so long... I don't see why we would wait for 4.2 to merge this... the way I see it... it isn't a breaking change at all...

@pxpm could you please give this a test & review, and we'll merge it in 4.1 if you also think this is ok to do?

**Review by pxpm — APPROVED** [review]
Thanks for the contribution @izica 

I've just tested this, and it works as expected!

**Comment by tabacitu** [comment]
Merged! Will be available later today, so starting with 4.1.53.

Thanks again @izica and sorry for the delay.

thank_you_cap

**Comment by izica** [comment]
I think now need to add this feature to the docs

**Comment by pxpm** [comment]
Totally agree @izica !

Before it slips under the table I've already created the PR.

Thanks! 
Pedro

**Pull request merged**